### PR TITLE
Stops you from activating sawflies in sanctuary zones

### LINKED ISSUE #12488: sawflies do not observe murderbox prep zone truce
### Labels

[BUG][MAJOR]

### Description

sawflies deploy in the murderbox prep zone and you can't do anything about it

### Steps to reproduce

1. be in murderbox prep area
2. buy sawflies
3. push the remote
4. get robusted by sawflies
5. you cannot robust the sawflies and die

### Expected Behavior

sawflies show love and kindness in more conventional ways

### Additional Information & Screenshots

Reported by: RDCB
On server: Goonstation 3 Roleplay: Morty
Round log date: 2022-12-23-16-35
Reported on: 2022-12-23 16:48:38
Map: NADIR

### PR DESCRIPTION
**Pull request by goonstation-enjoyer**
[OBJECTS]
## About the PR 
Title. You can still use them in the nukie ship.

I'm not sure if I should use 'notice' for the span class or something else

## Why's this needed? 
You probably shouldn't be able to activate them in sanctuary zones. You can't even fight back. :(
Fixes #12488

### CODE REVIEW

File: code/obj/sawflymisc.dm [hunk 1]
```diff
@@ -35,6 +35,13 @@ TYPEINFO(/obj/item/old_grenade/sawfly)
 	//used in dictating behavior when deployed from grenade
 	var/mob/living/critter/robotic/sawfly/heldfly = null
 
+	attack_self(mob/user)
+		var/turf/T =  get_turf(src)
+		var/area/A = T.loc
```
Comment (frawhst) [inline code/obj/sawflymisc.dm:40]: also `get_area()`

**Review by frawhst — COMMENTED** [review]
why are we letting people prime them on the battlecruiser anyways. I can't imagine an actual reason to do that other than misclicking.
Also, changelog is fine, but indicate that you can still use them in the murderbox probably.

**Comment by goonstation-enjoyer** [comment]
So syndies can mess with them before they use them onstation, if you'd like I can remove that part
Thank you, didn't know `get_area()` existed. <sup>(absolutely amazing now this is an XS pr)</sup>

File: code/obj/sawflymisc.dm [hunk 2]
```diff
@@ -35,6 +35,12 @@ TYPEINFO(/obj/item/old_grenade/sawfly)
 	//used in dictating behavior when deployed from grenade
 	var/mob/living/critter/robotic/sawfly/heldfly = null
 
+	attack_self(mob/user)
+		var/area/A = get_area(src)
+		if (A.sanctuary == TRUE && !istype(A, /area/syndicate_station/battlecruiser)) // salvager vessel, vr, THE SHAMECUBE, but not the battlecruiser
+			boutput(user, "<span class='notice'>You can't prime it here!</span>")
```
Comment (frawhst) [inline code/obj/sawflymisc.dm:41]:
```suggestion
			boutput(user, "<span class='notice'>You can't prime [src] here!</span>")
```
slightly cleaner (good to put the name of objects in messages for clarity)

**Review by frawhst — APPROVED** [review]

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**